plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/loops/SelfHyperLoopLabels.java

```diff
@@ -0,0 +1,257 @@
+    private PortSide side;
+    /** The label's alignment. */
+    private Alignment alignment;
+    /** The port any non-center alignment is relative to. */
+    private SelfLoopPort alignmentReferenceSLPort;
+    
+    
+    /**
+     * Creates a new instance for the given {@link SelfHyperLoop}.
+     */
+    public SelfHyperLoopLabels(final SelfHyperLoop slLoop) {
+        // Initialize our properties from the graph
+        LNode lNode = slLoop.getSLHolder().getLNode();
+        
+        layoutDirection = lNode.getGraph().getProperty(LayeredOptions.DIRECTION);
+        labelLabelSpacing = LGraphUtil.getIndividualOrInherited(lNode, LayeredOptions.SPACING_LABEL_LABEL);
+    }
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // LLabel Access
+    
+    /**
+     * Adds the given collection of labels to this instance.
+     */
+    public void addLLabels(final Collection<LLabel> newLLabels) {
+        for (LLabel newLLabel : newLLabels) {
+            lLabels.add(newLLabel);
+            updateSize(newLLabel);
+        }
+    }
+
+    /**
+     * Returns the labels represented by this instance.
+     */
+    public List<LLabel> getLLabels() {
+        return lLabels;
+    }
+    
+    /**
+     * Updates our size under the assumption that the given label was just added to this instance.
+     */
+    private void updateSize(final LLabel newLLabel) {
+        assert lLabels.get(lLabels.size() - 1) == newLLabel;
+        
+        KVector newLLabelSize = newLLabel.getSize();
+        
+        if (layoutDirection.isHorizontal()) {
+            // The labels will be stacked vertically
+            size.x = Math.max(size.x, newLLabelSize.x);
+            size.y += newLLabelSize.y;
+            
+            // Add a label-label spacing if we already had labels
+            if (lLabels.size() > 1) {
+                size.y += labelLabelSpacing;
+            }
+            
+        } else {
+            // The labels will be stacked horizontally
+            size.x += newLLabelSize.x;
+            size.y = Math.max(size.y, newLLabelSize.y);
+            
+            // Add a label-label spacing if we already had labels
+            if (lLabels.size() > 1) {
+                size.x += labelLabelSpacing;
+            }
+        }
+    }
+    
+    /**
+     * Applies label management with the given target width to our labels and updates our size. 
+     */
+    public void applyLabelManagement(final ILabelManager labelManager, final double targetWidth) {
+        size.set(LabelManagementProcessor.doManageLabels(
+                labelManager, lLabels, targetWidth, labelLabelSpacing, layoutDirection.isVertical()));
+    }
+    
+    /**
+     * Applies the bounding box placement to the individual labels.
+     */
+    public void applyPlacement(final KVector offset) {
+        if (layoutDirection.isHorizontal()) {
+            applyHorizontalPlacement(offset);
```

Comment:
I would have expected this method to be called `applyVerticalPlacement(...)`, as I expected it to denote the way in which the labels are placed. (Probably you thought about the placement when having a horizontal layout direction?!)

Reply:
Yes, that's what I thought about, but yeah, I understand how that can be confusing. I changed the method names to `applyPlacementForHorizontalLayout(...)` and `applyPlacementForVerticalLayout(...)`